src/components/DegreeProgressBar.tsx

```diff
@@ -0,0 +1,257 @@
+      <Grid
+        container
+        bgcolor="#FFDBAC"
+        width={`${majorCourse + reqElective + optElective + genEd}%`}
+        borderRadius="40px"
+      >
+        <Grid
+          container
+          bgcolor="#FFD700"
+          width={`${
+            ((majorCourse + reqElective + optElective) /
+              (optElective + reqElective + majorCourse + genEd)) *
+            100
+          }%`}
+          borderRadius="40px"
+        >
+          <Grid
+            container
+            width={`${
+              ((reqElective + majorCourse) /
+                (optElective + reqElective + majorCourse)) *
+              100
+            }%`}
+            bgcolor="orange"
+            borderRadius="40px"
+          >
+            <Grid
+              width={`${(majorCourse / (reqElective + majorCourse)) * 100}%`}
+              bgcolor="red"
+              borderRadius="40px"
+            ></Grid>
+          </Grid>
+        </Grid>
+      </Grid>
+    </Grid>
+  );
+}
+
+/**
+ * To-do dot labeling component constructor
+ * @returns {JSX.Element} - Dot labeling component for the to-do bar
+ */
+function DotLabelToDo(): JSX.Element {
+  return (
+    <Grid container justifyContent="center" mt="7px">
+      <Grid item container xs={5} alignItems="center" gap="3px">
+        <CircleRoundedIcon sx={{ color: 'red', fontSize: '11px' }} />
+        <Typography>Major Courses</Typography>
+      </Grid>
+
+      <Grid item container xs={5} alignItems="center" gap="3px">
+        <CircleRoundedIcon sx={{ color: 'orange', fontSize: '11px' }} />
+        <Typography>Required Electives</Typography>
+      </Grid>
+
+      <Grid item container xs={5} alignItems="center" gap="3px">
+        <CircleRoundedIcon sx={{ color: '#FFD700', fontSize: '11px' }} />
+        <Typography>Optional Electives</Typography>
+      </Grid>
+
+      <Grid item container xs={5} alignItems="center" gap="3px">
+        <CircleRoundedIcon sx={{ color: '#FFDBAC', fontSize: '11px' }} />
+        <Typography>GEs</Typography>
+      </Grid>
+    </Grid>
+  );
+}
+
+/**
+ * Degree progress bar component constructor
+ * @returns {JSX.Element} - The degree progress bar component
+ */
+export default function DegreeProgressBar(): JSX.Element {
+  const [isSwitchOn, setSwitchState] = useState(false);
+
+  const finishedPerc = 0;
+  const progressPerc = 0;
+
+  const mCourse = 0;
+  const reqElec = 0;
+  const optElec = 0;
+  const genEd = 0;
+
+  const handleChange = (): void => {
+    setSwitchState(!isSwitchOn);
+    console.log(isSwitchOn);
+  };
+
+  /**
+   * Information pop-up component constructor
+   * @returns {JSX.Element} - Information pop-up component
+   */
+  function InfoPopup(): JSX.Element {
+    const handleButtonClick = (): void => {
+      window.open(`https://idp.cpp.edu/idp/profile/cas/login?service=
+      https://cmsweb.cms.cpp.edu/psp/CPOMPRDM/EMPLOYEE/SA/c/POM_MENU_SA_SS.POM_SS_DPR_LINK.GBL`);
```

Comment:
So sorry I gave you the wrong link. Turns out the way cpp does authentication is a bit weird so for whatever reason adding a ?1=1 the end of the url makes it valid, while not having that makes it work only if you're already logged in depending on the page you try to access. 

So the full correct url should be:
https://idp.cpp.edu/idp/profile/cas/login?service=https://cmsweb.cms.cpp.edu/psp/CPOMPRDM/EMPLOYEE/SA/c/POM_MENU_SA_SS.POM_SS_DPR_LINK.GBL?1=1

Again no idea why cpp does this but in my testing ?1=1 seemed to work all the time, while removing it only worked sometimes. If you could verify this on your local machine please let me know if this link is invalid otherwise change the url accordingly.

Reply:
Also the url errors out when using a new line and `. Wrap the url with regular quotes instead and keep it one line.